NewForce-Cohort-6/tabloidfullstack-mac-cheese
# View a posts comments jj

This code addresses Ticket #19

[Link to Ticket #19 View A Posts Comments](https://trello.com/c/AjB635SH)

As a reader, I would like to see a list of all the Comments on a Post so that I can read and take part in the discussion on a particular Post.

*Given* the user is viewing the Details of a Post*When* they select the View Comments button*Then* they should be directed to the Comments list page for the Post*And* the list should be in order of creation date with the most recent on top*And* the title of the related Post should be displayed at the top of the page*And* a link back to the Post should be available

Display the following information for each Comment

* Subject* Content* Author's Display Name* Creation date (MM/DD/YYYY)

<hr>

## Testing Instructions

In your terminal, from your current branch:
* `git add` and `git commit` whatever you're working on
* `git checkout main`
* `git fetch -a` or `git fetch all` to get the branch
* If files do not show, `git pull origin ViewAPostsComments-JJ`
* `git checkout ViewAPostsComments-JJ`
* start the application with `npm start` in your terminal
* Ensure that you have created and are running the Tabloid database

### Data (if needed)

[Fill out the tables and columns](https://github.com/NewForce-Cohort-6/tabloidfullstack-mac-cheese/blob/main/SQL/01_Tabloid_Create_DB.sql)

[Seed data](https://github.com/NewForce-Cohort-6/tabloidfullstack-mac-cheese/blob/main/SQL/02_Tabloid_Seed_Data.sql)

### To Test

* Launch the api from VisualStudio using the filled in green 'play' button at the top.
* Lauch the front end code by right clicking the 'client' folder in visual studio solution explorer and selecting 'Open in terminal'
* Once the terminal opens, type 'code .'
* This should open visual studio code. Open a new terminal and type 'npm start'
* In the browser, if not logged in already, register an email with the registration button, then log in with that email
* Once logged in, click the posts button in the navbar at the top.
* verify that the url ends in `/posts`
* add `/1` to the end of the url so it looks like `/posts/1`
* You should see the details of the post, including the title "morph front-end markets", the Author "Rosana Yakushkev", and a place for comments.
* Under the details of the post, you should see a table for the comments. On this particular post, you should see a subject, content, display name, and Comment Date Time.
* If you wish, change the `/1` to other numbers to see comments on other posts.
